### ISSUE TITLE
New Wind/Pollution Rose dialogue

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#5965]
This is a new dialogue to implement the Wind/Pollution Rose plot from the OPenair package.  It is to go into the new Circular menu.
It is described in Chapter 5 of the Openair [Manual](https://davidcarslaw.com/files/openairmanual.pdf) together with examples of its use.
I start the design below, but there are many options which I hope that @HawardKetoyoMsatsi can explore to add, in particular to the sub-dialogue structure.  It can be merged in stages as facilities are implemented.

So it starts with the main dialogue having the same structure as the existing Wind Rose.  It is called `Wind/Pollution Rose`.  The main selector is the same as are the 2 single controls on the right, namely `Wind Speed: `and `Wind Direction:`.  
Under that is a checkbox called Compare.  Default is unchecked.  If checked then there are 2 more single receivers called `Wind Speed 2: `and `Wind Direction 2:`. 

Then (as in the existing wind rose plot) there is Facet (Optional).  Under that is Facet 2 (Optional):
The existing Wind Rose has the Titles, etc on the main dialogue.  Move this here to the Windrose Options dialogue here.
There is a `Windrose Options` button.
Of course the Save Graph control and the bottom Comment/OK control.
A checkbox under the options button `Paddle`.  Default is checked
A checkbox under that `Normalise`. Default is unchecked.
A drop-down with label `Summary`.  Options are Proportion (default), Mean, Count.

There is more to be put on the sub-dialogue.  But this could be merged (and tested) with these.

I hope @HawardKetoyoMsatsi can check the design, and consider the sub-dialogue, while @Wycklife

**Comment by rdstern** [comment]
@Ivanluv given Haward has moved (temporarily) to some comparison work with Danny, I wonder whether you could move to adding this dialogue?

**Comment by Ivanluv** [comment]
@rdstern  I will be haapy to take this up

**Comment by Ivanluv** [comment]
@HawardKetoyoMsatsi  has already added the windrose form in his PR #5970.I can continue working on this when the PR is merged

**Comment by lloyddewit** [comment]
@Ivanluv Just a note that PR #5970 has been replaced by PR #5983

**Comment by rdstern** [comment]
@HawardKetoyoMsatsi I have suggested you take over the coding of this example.  This is partly because it will become a very important dialogue for us, in at least 2 ways.  As the title of the dialogue makes clear, it allows us to start with tools for pollution data.  In addition, as a wind-rose plot is seems extremely flexible and powerful.

Unfortunately it also seems to be awkward to code.  This is because it has (clever) default names of ws for wind speed and wd for wind direction, and also some standard names for pollutants.  This makes the commands simpler for data frames with these special names, but it seems also to check for their existence, which often messes up the use when they do not exist.  @shadrackkibet noted one issue above, but I think the situation on special names is more complicated than that.  I have tried experimenting with the Niger data, (see #5997) where the elements are x58 for wind speed.  Wind direction is x59, but that must be multiplied by 10 to make it into 0 to 360.

Here is the dialogue I used:

![image](https://user-images.githubusercontent.com/11226469/94522906-4be24700-0228-11eb-9a58-21be75c11580.png)

Here are the data and the results:
![image](https://user-images.githubusercontent.com/11226469/94523072-9237a600-0228-11eb-95d5-df0159680a06.png)

Aha it may not be as awkward as I thought.  I see, from @shadrackkibet  comment above, that the new code uses pollutionRose, rather than windRose and it silently adds pollution = "ws" to the command.  So the problem is self-inflicted".  The command generated is:

`last_graph <- openair::pollutionRose(mydata=Niger10, ws="X58", wd="X59", paddle=TRUE, normalise=FALSE, statistic="prop.count", type="station", pollutant="ws") 
`
So it (silently) needs ws in the data set - which is a problem.  To correct, get rid of the `pollutant="ws"`  and change `pollutionRose` back to `windRose`.  

Then, I have a few more minor suggestions:
a) Make the first label Wind Speed (or Pollution): 
b) Add a checkbox (default unchecked) labelled Facets.  Only show the 2 facet variables if checked.
c) Move the facets to above the compare checkbox.
d) Correct the facet labels - add a space before (Optional) and Optional, not OPtional. 

Then let's see about the sub-dialogue.
